tests/ui/coverage-attr/allowed-positions.rs

```diff
@@ -0,0 +1,116 @@
+//! Tests where the `#[coverage(..)]` attribute can and cannot be used.
+
+//@ reference: attributes.coverage.allowed-positions
```

Comment:
cc @rust-lang/spec: should compiler reviewers ping you if a reference annotated test has expanded test coverage or modified test coverage?

Reply:
Actually don't answer that (sorry), I'll add that as a question for the T-spec testing RFC.